### PR TITLE
[docs] Bitbucket Pipelines CI documentation

### PR DESCRIPTION
**Pull request by dej611**
This PR adds some documentation on how to setup Testcafe in a new continuous integration as the Bitbucket Pipelines .
The Travis CI documentation and example have been used as baseline for this documentation.

Let me know if I need to change any formality or anything else in here.

### CODE REVIEW

**Comment by peteygao** [comment]
What are the flags `--no-sandbox --disable-setuid-sandbox` for? I have been using TestCafe with Chrome on Bitbucket Pipelines for a few months now, and I have never used either of those flags and I have not run into any issues. Can you explain what they are supposed to do? Disabling sandbox security mode seems like a bad idea in general (sure, it's running in an ephemeral docker container, so even if the browser and machine is compromised, it's no big deal... but it doesn't seem like an advisable thing to put into the documentation for _everyone_ to do).

**Comment by dej611** [comment]
Hi @peteygao,

I could not make it work without those flags. I have the same concerns as yours on the security but couldn't make it work without: the testcafe instance would just hang on initialisation step without starting and then exit on timeout.
If you could share part of your configuration to compare maybe we could sort that out and remove those two flags to covey better security also in the CI environment.

**Comment by AndreyBelym** [comment]
@peteygao, it's strange that you were able to run Chrome in a Docker container without disabling its sandbox, because AFAIK Chrome can't create a sandbox if the container doesn't have admin priveleges: [Running Puppeteer in Docker](https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker). Note the `--cap-add=SYS_ADMIN` flag in the `docker run` command. 

So unless BitBucket allows running containers with addtional privileges, we should disable Chrome's sandboxing by specifying `--no-sandbox --disable-setuid-sandbox`.

**Comment by VasilyStrelyaev** [comment]
@dej611, we really appreciate your efforts.

Yet, is there any particular reason why you did not use the [TestCafe Docker image](https://hub.docker.com/r/testcafe/testcafe/) in this tutorial? We generally recommend it if there are no factors that prevent this. This image is based on Alpine Linux and contains Node.js, TestCafe, Chromium and Firefox. This should be enough for the tutorial purposes - you won't need to install anything.

Before you start making changes (if you decide to do so), I'd like to draw your attention to the specifics connected with this approach. The CI system should run the `/opt/testcafe/docker/testcafe-docker.sh` script instead of the `testcafe` command to prepare the image for running browsers. You can use the [GitLab recipe](https://devexpress.github.io/testcafe/documentation/continuous-integration/gitlab.html) as an example. Yet, unlike GitLab, Bitbucket doesn't seem to require overriding the entrypoint - you can try to omit this setting.

At the same time, I believe we should leave your approach as an alternative option. We could start with a description of how to use the TestCafe image and then mention that users can also install TestCafe in a different image. Just as we do in the [GitLab recipe](https://devexpress.github.io/testcafe/documentation/continuous-integration/gitlab.html).

What do you think about this option?

**Comment by VasilyStrelyaev** [comment]
@dej611, I apologize for the delayed response. There was too much work with the upcoming v1.0.0 release.

I was going to give this doc a closer look and suggest specific changes.

Anyway, thank you for taking the initiative - I appreciate it a lot!

At first glance, the topic looks great, good job!

I wish I could review it in detail ASAP, but I'm afraid I won't be available for one more week.

I'm thinking of taking this pull request over from you and fixing some minor things by myself - I don't want to bother you with commas, indentation or word choice.

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+layout: docs
+title: Run Tests in Bitbucket Pipelines
+permalink: /documentation/continuous-integration/bitbucket-pipelines.html
+---
+# Run Tests in Bitbucket Pipelines CI
+
+You can automatically run TestCafe tests as a part of your build process on [Bitbucket Pipelines CI](https://bitbucket.org/product/features/pipelines).
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: You can run TestCafe tests as a part of your build process on [Bitbucket Pipelines CI](https://bitbucket.org/product/features/pipelines).

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Suppose you have a Bitbucket project for which you need to automatically run tests in the cloud when the project is modified. To do this, go through the following steps:
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: "Suppose you have a Bitbucket project for which you need to automatically run tests in the cloud when the project is modified. To do this, go through the following steps:" -> You can set up your Bitbucket project to automatically run tests in the cloud when the project is modified:

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Step 1 - Install TestCafe and Create Tests
+
+Install TestCafe [locally](../using-testcafe/installing-testcafe.md#locally) in your project and [create tests](../getting-started/README.md#creating-a-test).
+
+## Step 2 - Enable Bitbucket Pipelines for Your Project
+
+To enable Bitbucket Pipelines for your project, create a `bitbucket-pipelines.yml` file in the project's root folder.
+
+Bitbucket Pipelines CI allows you to avoid manual environment configuration and use Docker containers set up to work with Chrome and Firefox. You can [use a TestCafe Docker image](#option-1---use-testcafe-docker-image) with pre-installed software.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: avoid manual environment configuration?

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+However, if you already have a Docker image prepared to deploy your web application and run tests, you can [install TestCafe on this image before testing](#option-2---install-testcafe-on-a-docker-image).
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: However, if you already have a Docker image prepared, you can [install TestCafe on this image before testing](#option-2---install-testcafe-on-a-docker-image).

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md (hunk 3 above)
Comment (dirk-pieterse) [reply docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: Bitbucket Pipelines CI automatically configures the environment and you can use Docker containers that are set up for Chrome and Firefox ([use a TestCafe Docker image](#option-1---use-testcafe-docker-image) with pre-installed software).

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+For more information about how to get started with Pipelines, see the [Atlassian Bitbucket support page](https://confluence.atlassian.com/bitbucket/get-started-with-bitbucket-pipelines-792298921.html).
+
+### Option 1 - Use TestCafe Docker Image
+
+Specify the `testcafe/testcafe` image name in the `image` field of the [bitbucket-pipelines.yml file](https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html). Then, configure a pipeline that installs the project's dependencies and triggers a custom TestCafe launcher when you push commits and create pull requests.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: Specify the `testcafe/testcafe` image name in the `image` field of the [bitbucket-pipelines.yml file](https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html).

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            - /opt/testcafe/docker/testcafe-docker.sh firefox:headless,chromium tests/**/*
+```
+
+The custom launcher script `/opt/testcafe/docker/testcafe-docker.sh` prepares the environment to run a browser and starts TestCafe. Its arguments are standard TestCafe [command line parameters](../using-testcafe/command-line-interface.md).
+
+Commit and push this file to your repository.
+
+### Option 2 - Install TestCafe on a Docker Image
+
+Use the `image` field in the [bitbucket-pipelines.yml file](https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html) to specify a desired Docker image with Node.js and browsers installed. Then, configure a pipeline that installs the project's dependencies and runs tests when you push commits and create pull requests.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: Use the `image` field in the [bitbucket-pipelines.yml file](https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html) to specify a Docker image with Node.js and browsers installed.

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    master:
+      - step:
+          script:
+            - npm ci
+            - npm test
+```
+
+To specify how npm should run TestCafe, add the `test` script to the project's `package.json` file and execute the `testcafe` command in this script.
+
+The following example shows a command that runs tests in Chromium in headless mode.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: The following example shows a command that runs tests in Chromium in headless mode:

File: docs/articles/documentation/continuous-integration/bitbucket-pipelines.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+For more information on how to configure a test run with the `testcafe` command, see [Command Line Interface](../using-testcafe/command-line-interface.md).
+
+Commit the changes and push them to your repository.
+
+### Tip: Start a Custom Web Server
+
+If you need to start a custom Web server to host your application, use the [--app](../using-testcafe/command-line-interface.md#-a-command---app-command) TestCafe option followed by a command that starts this server.
+TestCafe executes this command automatically before tests are launched. After tests finish, TestCafe stops the server.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/continuous-integration/bitbucket-pipelines.md]: "TestCafe executes this command automatically before tests are launched." -> TestCafe executes this command before tests are launched.

File: examples/running-tests-in-chrome-using-bitbucket-pipelines-ci/README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# Example: Running Tests in Chrome Using Bitbucket Pipelines CI
+
+*A set of sample files that help you learn how to run TestCafe tests in the cloud using Chrome (Headless) with Bitbucket Pipelines CI.*
```
Comment (dirk-pieterse) [inline examples/running-tests-in-chrome-using-bitbucket-pipelines-ci/README.md]: help you learn?

Comment (dirk-pieterse) [reply examples/running-tests-in-chrome-using-bitbucket-pipelines-ci/README.md]: *A set of sample files that explains how to use Bitbucket Pipelines CI in Chrome (Headless) to run TestCafe tests in the cloud.*

File: examples/running-tests-in-chrome-using-bitbucket-pipelines-ci/README.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+## Running the example
+
+1. Clone the TestCafe repository to your machine.
+
+     ```sh
+     git clone https://github.com/DevExpress/testcafe.git
+     ```
+
+2. Create a new Bitbucket repository and copy the sample files from *examples/running-tests-in-chrome-using-bitbucket-pipelines-ci* to your repository.
+3. Go through steps described in the [Running Tests in Bitbucket Pipelines CI](http://devexpress.github.io/testcafe/documentation/recipes/running-tests-in-chrome-using-bitbucket-pipelines-ci.html) topic.
```
Comment (dirk-pieterse) [inline examples/running-tests-in-chrome-using-bitbucket-pipelines-ci/README.md]: Go through steps? Follow the steps described in the

**Pull request merged**